tests/integration/test_attach_backup_from_s3_plain/test.py

```diff
@@ -0,0 +1,40 @@
+# pylint: disable=global-statement
+# pylint: disable=line-too-long
+
+import pytest
+from helpers.cluster import ClickHouseCluster
+
+cluster = ClickHouseCluster(__file__)
+node = cluster.add_instance(
+    "node",
+    main_configs=["configs/disk_s3.xml"],
+    with_minio=True,
+)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def start_cluster():
+    try:
+        cluster.start()
+        yield
+    finally:
+        cluster.shutdown()
+
+
+def test_attach_backup():
+    node.query(
+        f"""
+    -- BACKUP writes Ordinary like structure
+    set allow_deprecated_database_ordinary=1;
+    create database ordinary engine=Ordinary;
+
+    create table ordinary.test_backup_attach engine=MergeTree() order by tuple() as select * from numbers(100);
+    -- NOTE: name of backup ("backup") is significant.
+    backup table ordinary.test_backup_attach TO Disk('backup_disk_s3_plain', 'backup');
+
+    drop table ordinary.test_backup_attach;
+    attach table ordinary.test_backup_attach (number UInt64) engine=MergeTree() order by tuple() settings storage_policy='attach_policy_s3_plain';
+    """
+    )
+
+    assert int(node.query("select count() from ordinary.test_backup_attach")) == 100
```

Comment:
What will happen if ClickHouse tries to merge parts of `test_backup_attach` ?

Reply:
It will throw some error, since it does not support mtime/move/...

Reply:
But merges will be assigned automatically, won't they? So it seems after a while such MergeTree table will spam in logs that it cannot do merging.

Reply:
Yes and this is like with `web` disk right now, eventually you will get some errors in the log, for proper support some "readonly" property of tables should be implemented.

And I completely agree with you that this should be done.

P.S. moreover there are random SIGSEGVs in web disk when you will try to do this and not only this.

Reply:
So to clarify, I'm suggesting fixing this separately (since there are other places to fix, like you should not do OPTIMIZE for read-only tables, and isReadOnly should be extended)

Reply:
Yes I agree, the current implementation must be useful, some issues can be fixed later.